lib/src/Utils/File.cpp

```diff
@@ -0,0 +1,119 @@
+// OpenCL SDK includes
+#include <CL/Utils/File.hpp>
+
+// STL includes
+#include <fstream>
+#include <iterator>
+#include <algorithm>
+#include <iostream>
+
+std::string cl::util::read_text_file(const char* const filename,
+                                     cl_int* const error)
+{
+    std::ifstream in(filename);
+    if (in.good())
+    {
+        try
+        {
+            std::string red((std::istreambuf_iterator<char>(in)),
+                            std::istreambuf_iterator<char>());
+
+            *error = CL_SUCCESS;
+            return red;
+
+        } catch (std::bad_alloc& ex)
+        {
+            (void)ex;
+            detail::errHandler(CL_OUT_OF_RESOURCES, error, "Bad allocation!");
+            return std::string();
+        }
+    }
+    else
+    {
+        detail::errHandler(CL_INVALID_VALUE, error, "No file!");
+        return std::string();
+    }
+}
+
+std::vector<unsigned char>
+cl::util::read_binary_file(const char* const filename, cl_int* const error)
+{
+    std::ifstream in(filename, std::ios::binary);
+    if (in.good())
+    {
+        try
+        {
+            std::vector<unsigned char> buffer(
+                std::istreambuf_iterator<char>(in), {});
+            return buffer;
+        } catch (std::bad_alloc& ex)
+        {
+            (void)ex;
+            detail::errHandler(CL_OUT_OF_RESOURCES, error, "Bad allocation!");
+            return std::vector<unsigned char>();
+        }
+    }
+    else
+    {
+        *error = CL_INVALID_VALUE;
+        return std::vector<unsigned char>();
+    }
+}
+
+
+cl::Program::Binaries
+cl::util::read_binary_files(const std::vector<cl::Device>& devices,
+                            const char* const program_file_name,
```

Comment:
If program_file_name is only a base file name, perhaps we should indicate that in the variable name. IntelliSense will confuse people with this variable name _and_ no Doxygen summary popping up.